client/app/react-components/QueryViewMain.js

```diff
@@ -0,0 +1,312 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect, PromiseState } from 'react-refetch';
+import Mustache from 'mustache';
+import { includes, some, union, uniq } from 'lodash';
+import moment from 'moment';
+
+import FlexResizable from './FlexResizable';
+import QueryViewNav from './QueryViewNav';
+import QueryViewVisualizations from './QueryViewVisualizations';
+import QueryViewFooter from './QueryViewFooter';
+import QueryEditor from './QueryEditor';
+import QueryMetadata from './QueryMetadata';
+
+
+function collectParams(parts) {
+  let parameters = [];
+
+  parts.forEach((part) => {
+    if (part[0] === 'name' || part[0] === '&') {
+      parameters.push(part[1]);
+    } else if (part[0] === '#') {
+      parameters = union(parameters, collectParams(part[4]));
+    }
+  });
+
+  return parameters;
+}
+
+function parseQuery(query) {
+  return uniq(collectParams(Mustache.parse(query)));
```

Comment:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_uniq

among others. I won't call out every example, but that GitHub page is a good resource.